b2sdk/version.py

```diff
@@ -18,3 +23,178 @@
 PYTHON_VERSION = '.'.join(map(str, sys.version_info[:3]))  # something like: 2.7.11
 
 USER_AGENT = 'backblaze-b2/%s python/%s' % (VERSION, PYTHON_VERSION)
+
+
+class Deprecat:
+    def _get_verified_versions(self, tuple_, subject, from_=None, allow_future=False):
+        """
+        subject may be "function", "method", "argument", "class" etc
+        returns None if replacement is going to happen in the future, or a 3-tuple with replacemnt specification:
+            * name to replace to
+            * version where the change was introduced
+            * version where the support for the old name is going to be dropped
+        """
+        to_, version_changed, version_dropped = tuple_
+        assert len(version_changed) == len(version_dropped) == len(
+            VERSION
+        ) == 5  # TODO the comparator used below is too naiive for the provided version number
+        assert version_changed < version_dropped, '%s decorator is set to start renaming %s %r starting at version %s and finishing in %s. It needs to start at a lower version and finish at a higher version.' % (
+            self.__class__.__name__, subject, from_, version_changed, version_dropped
+        )
+        assert VERSION < version_dropped, '%s decorator is still used in version %s when old %s name %r was scheduled to be dropped in %s. It is time to remove that mapping.' % (
+            self.__class__.__name__, VERSION, subject, from_, version_dropped
+        )
+        if not allow_future:
+            assert version_changed <= VERSION, '%s decorator indicates that the replacement of %s %s should take place in the future version %s, while the current version is %s' % (
```

Comment:
`decorator indicates that the replacement of %s %s ` should that that a to in between `subject` and `from_`

Reply:
all messages were changed